### PR TITLE
Configuration 

### PR DESCRIPTION
**Pull request by hariso**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,14 @@
...
+The S3-Apache Iceberg connector is a [Conduit](https://github.com/ConduitIO/conduit) destination connector, 
+which makes it possible to write data into S3 in the [Apache Iceberg](https://iceberg.apache.org/) format.
+
+### Pre-requisites
+* JDK 17
+* Currently, only Unix-like OSes are supported.
+
+## Usage
+First, the connector needs to be built, which can be done with `scripts/dist.sh`. A new directory, `dist`,
+will be created. The contents of `dist` need to copied into the Conduit connectors' directory (which,
```
Comment (maha-hajja) [inline README.md:12]:
```suggestion
will be created. The contents of `dist` need to be copied into the Conduit connectors' directory (which, 
```

File: scripts/dist.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+mvn clean package
+
+TO_DIR=dist/
+rm -rf $TO_DIR
+mkdir -p $TO_DIR
+cp scripts/conduit-connector-s3-iceberg $TO_DIR
```
Comment (maha-hajja) [inline scripts/dist.sh:7]: simple yet super helpful script!

File: scripts/dist.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+cp target/conduit-connector-s3-iceberg-*.jar $TO_DIR/libs
```
Comment (maha-hajja) [inline scripts/dist.sh:8]: maybe we should create the `/libs` directory first? I tried running the script and "libs" was a runnable file, not a folder, maybe I did sth wrong not sure

File: src/main/java/io/conduit/DestinationConfig.java [hunk 4]
```diff
@@ -17,32 +17,103 @@
...
+    private Map<String, String> catalogProperties = new HashMap<>();
+
     /**
      * Creates a new <code>DestinationConfig</code> instance from a map with configuration parameters.
+     * The map is required to have:
+     * <li><code>namespace</code></li>
+     * <li><code>table.name</code></li>
+     * <li><code>catalog.name</code></li>
+     * <p>
+     * The catalog properties need to prefixed with <code>catalog.catalog_name</code>.
```
Comment (maha-hajja) [inline src/main/java/io/conduit/DestinationConfig.java:77]:
```suggestion
     * The catalog properties need to be prefixed with <code>catalog.catalog_name</code>.
```

File: src/main/java/io/conduit/DestinationConfig.java [hunk 5]
```diff
@@ -17,32 +17,103 @@
...
 
-        map.forEach(connectorMap::put);
-        T cfg = Utils.mapper.convertValue(connectorMap, clazz);
+        if (!Utils.isEmpty(missing)) {
+            throw new IllegalArgumentException("missing keys: " + missing);
+        }
+    }
 
-        return cfg;
+    // Capture all other fields that Jackson do not match other members
```
Comment (maha-hajja) [inline src/main/java/io/conduit/DestinationConfig.java:109]: do you mean Json? I'm not sure I understand this comment tbh

File: src/main/java/io/conduit/SpecifierService.java [hunk 6]
```diff
@@ -41,27 +41,48 @@ public class SpecifierService extends SpecifierPluginGrpc.SpecifierPluginImplBas
...
+            "table.name",
+            Specifier.Parameter.newBuilder()
+                .setDescription("Table name")
+                .setDefault("")
+                .setType(Specifier.Parameter.Type.TYPE_STRING)
+                .addValidations(requiredValidation)
+                .build()
+        );
+
+        // todo make Catalog URI required
```
Comment (maha-hajja) [inline src/main/java/io/conduit/SpecifierService.java:85]: what about the s3 params? shouldn't we have them here too?

Comment (hariso) [reply src/main/java/io/conduit/SpecifierService.java:85]: You're right!

File: scripts/dist.sh (hunk 3 above)
Comment (hariso) [reply scripts/dist.sh]: done

File: scripts/dist.sh (hunk 2 above)
Comment (hariso) [reply scripts/dist.sh:7]: thanks!

File: src/main/java/io/conduit/DestinationConfig.java (hunk 5 above)
Comment (hariso) [reply src/main/java/io/conduit/DestinationConfig.java]: removed comment

**Review by maha-hajja — APPROVED** [review]
LGTM!

**Pull request merged**